Detectors/MUON/MCH/DigitFiltering/src/DigitFilteringSpec.cxx

```diff
@@ -0,0 +1,169 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "MCHDigitFiltering/DigitFilteringSpec.h"
+
+#include "DataFormatsMCH/Digit.h"
+#include "DataFormatsMCH/ROFRecord.h"
+#include "Framework/ConfigParamRegistry.h"
+#include "Framework/DataSpecUtils.h"
+#include "Framework/Logger.h"
+#include "Framework/OutputSpec.h"
+#include "Framework/Task.h"
+#include "Framework/WorkflowSpec.h"
+#include "MCHBase/SanityCheck.h"
+#include "MCHDigitFiltering/DigitFilterParam.h"
+#include "SimulationDataFormat/MCCompLabel.h"
+#include "SimulationDataFormat/MCTruthContainer.h"
+#include <fmt/format.h>
+#include <iostream>
+#include <string>
+#include <vector>
+
+using namespace o2::dataformats;
+using namespace o2::framework;
+
+namespace o2::mch
+{
+class DigitFilteringTask
+{
+ public:
+  DigitFilteringTask(bool useMC) : mUseMC{useMC} {}
+
+  void init(InitContext& ic)
+  {
+    mSanityCheck = ic.options().get<bool>("sanity-check");
+    mMinADC = ic.options().get<int>("min-adc-value");
```

Comment:
Not sure this mix between device options and configurable param will always work as expected. I would remove the device options and set the values from the configurable param here.

Not sure it is useful but you can also add a device option --mch-config to load them from a INI or JSON file first if you want.

Reply:
Well, I wanted to minimise the number of changes and keep the outside usage unchanged. But I agree the mixing is probably not a good idea

Reply:
will keep the addition of the `--mch-config` option for later...